### PR TITLE
Add priv rand

### PR DESCRIPTION
**Pull request by richsalz**
This is based on #4019; only the last two commits are relevant.  (I'll rebase #4109 is reviewed+merged.)

This adds a separate global DRBG instance for generating private keys.  That way, even if the main DRBG is compromised, the DRBG instance for generating long-term private keys is not.  This "dual DRBG" approach is used by amazon, and a recommended practice.

### CODE REVIEW

**Comment by paulidale** [comment]
I like this a lot.  A separation between normal randoms and long lived randoms is good.

One thought: would it be worth considering the addition of some random bits for each `RAND_priv_bytes` request?  If a complete reseed isn't viable, even a few bits will provide some distinction.  I'd mostly consider sources that will return something very quickly.  RDRAND is the obvious one and provides a complete reseed if desired.  If it isn't available `clock_gettime(2)` and RDTSC could also help.  Even low randomness things like process CPU time and system memory status could provide some differentiation.

File: doc/man3/RAND_bytes.pod [hunk 1]
```diff
@@ -22,9 +23,15 @@ RAND_bytes() puts B<num> cryptographically strong pseudo-random bytes
...
 
+RAND_priv_bytes() has the same semantics as RAND_bytes().  It is intended to
+be used for generating long-term private keys. If using the default
+RAND_METHOD, this function uses a separate instance of the PRNG so that
+a compromise of the global generator will not affect such key generation.
+
 =head1 RETURN VALUES
 
-RAND_bytes() returns 1 on success, -1 if not supported by the current
+RAND_bytes() nad RAND_priv_bytes()
```
Comment (paulidale) [inline doc/man3/RAND_bytes.pod]: s/nad/and/

Comment (richsalz) [inline doc/man3/RAND_bytes.pod]: fixed; updated commit pushed.

**Comment by richsalz** [comment]
The instantiate function gets called by rand-bytes if needed; it's not seeded automatically at startup.  The idea of reseeding more often, and perhaps with fewer bytes, is a good idea.  Maybe after these other PR's get reviewed and merged.... hint, hint.

**Comment by paulidale** [comment]
Yeah, another one for my list of things to improve :)

**Comment by richsalz** [comment]
squashed and merged.

**Pull request closed without merging**